### ISSUE TITLE
The mask related code in the Dice loss function is wrong

### ISSUE THREAD

**Issue by nikolakopoulos** [ShannonAI/dice_loss_for_NLP#8]
Hello,

I found the following bug in your code:

        if mask is not None: # here is the problem!! Flat input is already made one-hot, thus the multiplication will not work!
            mask = mask.float()
            flat_input = flat_input * mask
            flat_target = flat_target * mask
        else:
            mask = torch.ones_like(target)
            
An easy fix is the following:

        if mask is not None: # here is the problem!! Flat input is already made one-hot, thus the multiplication will not work!
            mask = mask.float()
            flat_input = (flat_input.t() * mask).t()
            flat_target = (flat_target.t() * mask).t()
        else:
            mask = torch.ones_like(target)
